PictureViewMaster/PictureMasterView/PictureMasterViewController.swift

```diff
@@ -246,11 +261,35 @@ public class PictureMasterViewController: UIViewController , UIGestureRecognizer
     private func resetViewFrame(view: UIView, animated: Bool, completion: ((Bool) -> Void)?) {
         let duration = animated ? 0.2 : 0.0
         UIView.animateWithDuration(duration, delay: 0.0, options: .CurveEaseOut, animations: {
-            let imageView = view as! UIImageView
-            view.frame = self.originalImageViewFitFrameForImage(imageView.image!)
+            let imageView = view
+            view.frame = self.minimumSizeForView(imageView)
+//              view.frame = self.originalImageViewFitFrameForImage(imageView.image!)
             }, completion:completion)
     }
     
+    private func minimumSizeForView(view: UIView) -> CGRect {
+        let screenSize = UIScreen.mainScreen().bounds
+        var frame = CGRect()
+        
+        let heightRatio = view.frame.size.height / view.frame.size.width
+        let widthRatio = view.frame.size.width / view.frame.size.height
+        let screenRatio = screenSize.width / screenSize.height
+        
+        if view.frame.size.height > view.frame.size.width && widthRatio < screenRatio {
+            
+            frame.size.height = screenSize.height
+            frame.size.width = frame.height * widthRatio
+        }else {
+            frame.size.width = screenSize.width
+            frame.size.height = frame.width * heightRatio
+        }
+        
+        frame.origin.x = (screenSize.size.width / 2) - (frame.size.width / 2)
+        frame.origin.y = (screenSize.size.height / 2) - (frame.size.height / 2)
+        
+        return frame
```

Comment:
return `CGRect(....)` using your `size` and `origin` constants I suggested above.